content/blog/2018-05-02-onboarding-is-work.md

```diff
@@ -0,0 +1,325 @@
+involves a lot of work from many actors: authors, reviewers and editors;
+but *how much work*? Managing the effort involved in the peer-review
+process is a major part of ensuring its sustainability and quality. In
+this post, we’ll take a look the effort put in by participants in the
+review process, and also learn something about exploring GitHub data
+along the way.
+
+### Work done by authors
+
+We can try to quantify the development work of authors by looking at the
+number of lines deleted and added in the git repo of the package before,
+during, and after review. To do so, we link information from the package
+git repos (information about sizes and dates of commits) and information
+from the onboarding issue threads (start and end dates of the review
+process, as measured by open/close dates).
+
+``` r
+library("ggplot2")
+library("magrittr")
+
+commits <- readr::read_csv("output/gitsum_reports.csv")
+issues <- readr::read_csv("data/clean_data.csv")
+issues <- dplyr::group_by(issues, package) %>%
+  dplyr::summarise(opened = min(created_at),
+                   closed = max(closed_at))
+commits <- dplyr::left_join(commits, issues, by = "package")
+
+custom_transf <- function(x){
+  pof <- log(x)
+  pof[x == 0] <- 0
+  pof
+}
+
+custom_exp <- function(x){
+  pof <- exp(x)
+  pof[x < 0] <- - exp(-x)
+  pof
+}
+
+
+plot_commits <- function(package_name, commits){
+  message(package_name)
+  example <- dplyr::filter(commits, package == package_name, !is_merge)
+
+  p <- ggplot(example)   +
+    geom_segment(aes(x = datetime, xend = datetime,
+                     y = 0, yend = custom_transf(total_insertions)),
+                 col = "salmon", size = 1) +
+    geom_segment(aes(x = datetime, xend = datetime,
+                     y = 0, yend = - custom_transf(total_deletions)), 
+                 col = "blue", size = 1) +
+    ylab("lines") +
+    annotate("segment", 
+             x = min(example$datetime),
+             xend = max(example$datetime),
+             y = 0, yend = 0,
+             col = "grey50",
+             size = 0.2) +
+    scale_y_continuous(breaks= seq(min(-custom_transf(example$total_deletions)),
+                                   to = max(custom_transf(example$total_insertions)),
+                                   by = 5), 
+                       labels= round(custom_exp(seq(min(-custom_transf(example$total_deletions)),
+                                         to = max(custom_transf(example$total_insertions)),
+                                         by = 5))))+
+    annotate("rect",
+             xmin = example$opened[1], xmax = example$closed[1],
+             ymin = - custom_transf(max(example$total_deletions)), 
+             ymax = custom_transf(max(example$total_insertions)),
+             fill = "grey80", alpha = 0.6,
+             col = "grey10",
+             size = 0.2) +
+    ggtitle(package_name) +
+    hrbrthemes::theme_ipsum() 
+  print(p)
+}
+```
+
+Let’s have a look at a few packages.
```

Comment:
Would help to have some text to give overview of the figs of commit histories to indicate meaning of grey box, red and blue lines, before jumping to the interpretation. I can surmise it but an overview would be super helpful

Reply:
very good point! I actually still know by heart the explanations I gave in my talk, now I'm wondering why I did not add them right away to the post...